src/Storages/HDFS/ReadBufferFromHDFS.cpp

```diff
@@ -22,8 +22,6 @@ ReadBufferFromHDFS::~ReadBufferFromHDFS() = default;
 
 struct ReadBufferFromHDFS::ReadBufferFromHDFSImpl : public BufferWithOwnMemory<SeekableReadBuffer>
 {
-    /// HDFS create/open functions are not thread safe
-    static std::mutex hdfs_init_mutex;
```

Comment:
but it is just removing the mutex and leaving not thread-safe functions unprotected

Reply:
The thread unsafe code I found in ReadBufferFromHDFS is setenv/getenv LIBHDFS3_CONF and i think the mutex is for it.（Any other code unsafe？）
Using std::call_once for setenv is more efficient than the current static mutex. (getenv no need to protect)

Reply:
The thread unsafe code I found in ReadBufferFromHDFS is setenv/getenv LIBHDFS3_CONF and i think the mutex is for it.（Any other code unsafe？）
Using std::call_once for setenv is more efficient than the current static mutex. (getenv no need to protect)

Reply:
@alesapin  Could you help to review？ The mutex was added in #15675

Reply:
> The thread-unsafe code I found in ReadBufferFromHDFS is setenv/getenv LIBHDFS3_CONF and i think the mutex is for it.（Any other unsafe？）

As the comment states: it is hdfs functions which are not thread safe, in libhdfs3 library